### PR TITLE
Deathtrauma Improved

### PR DESCRIPTION
**Pull request by FriendlyContractor**
## About The Pull Request
Adds cell necrosis brain trauma which is applied on death. Cell necrosis will randomly cause permanent brain traumas to the person over time unless they take special medicine forcing them to stay on medication or risk sever consequences. Taking your medication however has various side effects and additionally taking larger doses cause them to be more frequent making for interesting choice between convinience of taking less frequent doses and dealing with worse side effects or taking more frequent doses and dealing with less side-effects. Adds new implant - brain cell stimmulator. Brain cell stimmulator decrases the risk of getting brain traumas from cell necrosis considerably decrasing the risk of going without medication when you are unable to or unwilling because of the side-effects. Mannitol will also slightly decrase the chance of random traumas as well while in bloodstream

## New medicine
All of them prevent random traumas from brain cell necrosis. Difference is in side effects and metabolization rate as well as complexite of recipes
* **Brendol( synaptizine, mannitol, oil )**
The easiest to make requiring only things from chem dispenser. Side effects include random stuns and less frequent random flashing. If overdosed(40u) causes significant brain damage and rarely clone damage. Metabolises slowly and full dose will last for some time
* **Cortexone( brendol, psicodine, saline-glucose, 2 plasma)**
Second tier medicine. More difficult to make as it requires plasma unavailable from dispenser. Takes almost twice as long to metabolize as brendol making it a worthwhile upgrade. Side effects include frequently droping items, sometimes random stamina damage and rarely falling asleep for a short amount of time.
* **Cranizine( cortexone, omnizine, frostoil, liquid adamantine, pax)**
Provides unrivaled quality of life however is difficult to make requiring cooperation with botany and xenobiology. Has no random side effects and instead pacifies the user and makes him hypersensitive to pain meaning you have to be extremely carefull to avoid damage as even small amounts will quickly cause you to be paralysed with pain

## Why It's Good For The Game

Adds consequences to death making it more significant event and causes players to care about their lives more. Adds interesting and fairly complex game mechanics around it adding new things for player to do and new choices to make

## Changelog
:cl:
add: Dying will now cause permanent cell necrosis brain trauma. Cell necrosis causes random permanent traumas over time unless person has special medicine in their bloodstream. Adds 3 new  reagents - brendol, cortexone and cranizine. They will prevent brain cell necrosis from causing traumas however they have various side effect which additionally get more severe in large doses. New implant has been added to help combat brain cell necrosis - brain cell stimmulator. When implanted in users head halves the chance person will receive random brain trauma making it uselfull if you go off medication. You can also take mannitol to slighty reduce chance of random traumas happening
/:cl:

### CODE REVIEW

**Comment by zeskorion** [comment]
the original idea was terrible. this is even worse.

**Comment by FriendlyContractor** [comment]
Would you elaborate on why in your opinion this is a terrible idea ?

**Comment by PerishedFraud** [comment]
I will now elaborate on why the idea is worse:

>Adds cell necrosis brain trauma which is applied on death. Cell necrosis will randomly cause permanent brain traumas to the person over time unless they take special medicine forcing them to stay on medication or risk sever consequences. Taking your medication however has various side effects and additionally taking larger doses cause them to be more frequent making for interesting choice between convinience of taking less frequent doses and dealing with worse side effects or taking more frequent doses and dealing with less side-effects. Adds new implant - brain cell stimmulator. Brain cell stimmulator decrases the risk of getting brain traumas from cell necrosis considerably decrasing the risk of going without medication when you are unable to or unwilling because of the side-effects. Mannitol will also slightly decrase the chance of random traumas as well while in bloodstream

>New medicine
All of them prevent random traumas from brain cell necrosis. Difference is in side effects and metabolization rate as well as complexite of recipes

>Brendol( synaptizine, mannitol, oil )
The easiest to make requiring only things from chem dispenser. Side effects include random stuns and less frequent random flashing. If overdosed(40u) causes significant brain damage and rarely clone damage. Metabolises slowly and full dose will last for some time
Cortexone( brendol, psicodine, saline-glucose, 2 plasma)
Second tier medicine. More difficult to make as it requires plasma unavailable from dispenser. Takes almost twice as long to metabolize as brendol making it a worthwhile upgrade. Side effects include frequently droping items, sometimes random stamina damage and rarely falling asleep for a short amount of time.
Cranizine( cortexone, omnizine, frostoil, liquid adamantine, pax)
Provides unrivaled quality of life however is difficult to make requiring cooperation with botany and xenobiology. Has no random side effects and instead pacifies the user and makes him hypersensitive to pain meaning you have to be extremely carefull to avoid damage as even small amounts will quickly cause you to be paralysed with pain
Why It's Good For The Game
Adds consequences to death making it more significant event and causes players to care about their lives more. Adds interesting and fairly complex game mechanics around it adding new things for player to do and new choices to make

>Changelog
🆑
add: Dying will now cause permanent cell necrosis brain trauma. Cell necrosis causes random permanent traumas over time unless person has special medicine in their bloodstream. Adds 3 new reagents - brendol, cortexone and cranizine. They will prevent brain cell necrosis from causing traumas however they have various side effect which additionally get more severe in large doses. New implant has been added to help combat brain cell necrosis - brain cell stimmulator. When implanted in users head halves the chance person will receive random brain trauma making it uselfull if you go off medication. You can also take mannitol to slighty reduce chance of random traumas happening

**Comment by mcmeiler** [comment]
...And how exactly is this better than the old PR? We've established that random permanent traumas because you died are bad.

**Comment by FriendlyContractor** [comment]
Because they are no longer unpreventable and will only happen if you neglect taking medication. This adds more stuff to do for medical and interesting choices for players to make such deciding the dose of medicine they wish to take considering side effect intensity and perhaps even tacticaly putting off medication completly for some time taking your chance with traumas when you absolutely cannot afford to suffer the side effects. Can be summed up as a change from randomly becoming permanently paralyzed because you died to having to carry a bottle of pills with you and occasionaly suffering moderatly annoying medicine side effects

**Comment by BrumalLake** [comment]
death is uncontrollable and this is a prick in the ass
why should we play while thinking of the future consequences
people will continue to dump their consequences on their future selves and feel like shit later while remembering the name FriendlyContractor

**Comment by Kapu1178** [comment]
can i veto this using Carbon Module Freeze as my reasoning

**Comment by mcmeiler** [comment]
> if you neglect taking medication

Lmao as if chem even bothers to make said medication



> when you absolutely cannot afford to suffer the side effects

Current side effects include:
- Random 40 stamina damage
- Random item dropping
- Random flash effects
- Extremely scaling stamina damage if you're over 10 brute and burn combined

*absolutely nobody is going to* ***want*** *to deal with these, ever, ever,* ***ever***
Remember how most jobs have to hold tools to, you know, do their job?

> moderatly annoying medicine side effects

Going into stamcrit randomly is not moderately annoying. It is *extremely dangerous for you and removes all fun from gameplay,* especially considering how you can essentially be in stamcrit for a good 20 minutes if you're taking a full size pill (Which is the only size chemsists would ever make) and get unlucky.

**Comment by DrMacCool** [comment]
The only way I've seen consequences for death working is scrapping revivals in any form and instead having respawns. It's actually my preferred way of handling death, but it would be terrible for Beestation, where the rounds can be quite short and you only get a handful of player slots. Also meta-grudging. Beestation has a really bad case of meta-relationships.

**Comment by francinum** [comment]
How did you get 'I should make this random trauma thing be able to proc more than once unless they actively prevent it' from anything that happened in the last PR.

**Comment by ilikesaying** [comment]
Die once from anything that may be out of your control
now you start getting _**permanent**_ brain traumas if medbay isn't making medicine or is gone entirely (bombed)
better hope you don't fucking die at all because you'll start becoming more and more of a vegetable
why bother getting revived
might as well be a borg or play a spawner

**Comment by RDS88-beeman** [comment]
Alright two things. One: Did you account for IPC's, who can't take medication and are easy to revive as an upside, and for whom cell necrosis makes no fucking sense on account of being fully mechanical. Because it doesn't look like you did.
And second of all: there are multiple typos in the description of Cranizine, or at least grammatical errors. I forgot how to leave reviews so I'll just quote it.
It is currently this: 

> description = "Allows living comfortable live even with brain cell necrosis but considerable affects user making him hypersensitive to pain and very docile"

when it should be this
> description = "Allows living a comfortable life even with brain cell necrosis but considerably affects the user making them hypersensitive to pain and very docile"

**Comment by RDS88-beeman** [comment]
Now that I've pointed out the technical issues an untrained eye can see, I can diss this PR.
Death is out of the players control most of the time, and punishing them for the rest of the round is generally a bad idea, as it effectively boils down to punishing players randomly. Now sure, it can be prevented, but for certain people you're better off just not fucking taking the meds and dealing with the traumas because the side effects are that bad.

This could work on a more RP oriented code base, where people actually do their jobs more and death is actually something to be feared, but /TG/ code is most comparable to an arcade. Quick in, quick out, with more focus placed on fighting the antagonists with rewards from the job rather than enjoying the job itself.

As such, chemists are prone to make narcotics, or put little effort into nonbasic medical chemicals, rather than diligently preparing every little chemical that anyone needs. That or they've already blown up, and are using all the chemicals for  themselves to deal with what would be a minor inconvenience before.

If death should be more impactful is a contentious issue, but revival being easy is a symptom of the fragility of life, not the fundamental problem. Bay, a code base where this might actually work, has pain, more durable humans, and _players that are all but guaranteed to actually RP a hostage situation as opposed to just gunning you down even when you surrender._ 

On those kinds of codebases, the thought goes
"I should capture this dude instead of killing them, as it gives me a bargaining chip to work with security, and round removing someone for breathing near me is just kind of mean. This is a more fun option"
on this codebase, the thought is
"No point in capturing, I'd have to lug them around and if I can't convert them odds are they just club me over the head as soon as they get out of the cuffs. Sec will kill me anyway, so there's no point bargaining, and round will be over in 45 minutes or they'll get revived, so they'll get to play again soon enough, if I even care about that kind of thing."

These points do tend to generalize the playerbases of both server types somewhat, but I think it's fair to say the "issue" this PR is trying to solve won't be fixed by making death more impactful first. Make dying more meaningful, something you have to go out of your way to do rather than the best option to disable someone in a fight. That's what this codebase was built around.

90% of traitor options are lethal, the second best thing to critting someone is the rare stamina weapon (which are usually heavily restricted to sec), the third was slipping or flashes, but powergaming rules prevents their use, so if someone comes at you and you get the upper hand, you get presented with a simple choice of kill or die. Death being less impactful makes both of these choices less frustrating, and if someone **really** hates you for whatever reason or needs you to stay dead for an objective, there are no shortage of permanent round removal solutions for the aspiring traitor (cremator, gib, deepfry brain, space brain, hide brain in a potted plant, etc etc)

TL;DR dying is the prime way of temporarily taking someone out for the aspiring traitor or random crewmate on self defense, and making that more difficult to undo won't make it any less required and will only hurt the gameplay experiences of those on the receiving end. The chems do nothing to help because chemists won't make them, which is a more fundamental flaw with both med and med players, and like 15% of the time they should, they don't even work. Also sometimes you just die because you made a minor mistake, or for no fault of your own, and making that more punishing is an even worse idea.

**Comment by DrMacCool** [comment]
I could only see this method being viable if this necrosis is a temporary debuff. Maybe make it so you have some kind of specific trauma that clears up after a short while, with the medications speeding up the recovery. But even then, being dead is the temporary debuff that comes with, well, being dead, so the point is moot.

**Comment by FriendlyContractor** [comment]
> _absolutely nobody is going to_ _**want**_ _to deal with these, ever, ever,_ _**ever**_ Remember how most jobs have to hold tools to, you know, do their job?

Of course nobody is going to want to deal with these those are not meant to be desirable effects. If you have a job which involves holding items for a prolonged time remember putting off medication for some time and taking other medicine instead are an options.

> Going into stamcrit randomly is not moderately annoying. It is _extremely dangerous for you and removes all fun from gameplay,_ especially considering how you can essentially be in stamcrit for a good 20 minutes if you're taking a full size pill (Which is the only size chemsists would ever make) and get unlucky.

This is simply untrue. Random stamina damage from cortexone is rare enough that it will almost never stack to cause a stamina crit even on maximum dose. Cranizine on the other hand can cause simmiliar effect to what you are describing but its a trade off to having no random annoyances other medicine has. With cranizine you have to change the way you play and be very carefull not to end up alone and be very skittish around potential sources of damage.

**Comment by FriendlyContractor** [comment]
> How did you get 'I should make this random trauma thing be able to proc more than once unless they actively prevent it' from anything that happened in the last PR.

This completly changes everything. Random traumas are now only a thing which will realisticaly occur if player intentionally chooses not to take easily prepared medication. Most players will never experience permanent traumas with the changes i propose compared to previous PR

**Comment by FriendlyContractor** [comment]
> 
> 
> Die once from anything that may be out of your control now you start getting _**permanent**_ brain traumas if medbay isn't making medicine or is gone entirely (bombed) better hope you don't fucking die at all because you'll start becoming more and more of a vegetable why bother getting revived might as well be a borg or play a spawner

Dying may be out of your control but taking medication is not. In realistic situation random brain traumas will never occure unless you intentionaly go off medication for one reason or another. If medbay is neglecting their duties or non-existant you are probably not gonna be revived in the first place and don't have to worry about random traumas either way

**Comment by Mick1299** [comment]
> Required to take medicine to prevent effect
> Starts when you die, never stops
> Incurable

Sounds like a permanent brain trauma to me

**Comment by mcmeiler** [comment]
This is literally worse, as was said before, it's *literally* the last PR but it can proc multiple times. Chemistry won't make meds, the amount in vendors is insignificant, and constantly being on horrible side effects is awful and not fun to deal with.

**Comment by imsynthex** [comment]
the revive process itself should be harder

there shouldnt be irreversible traumas that you get from death, the idea is bad no matter how you implement it

**Comment by ilikesaying** [comment]
> Dying may be out of your control but taking medication is not. In realistic situation random brain traumas will never occure unless you intentionaly go off medication for one reason or another. If medbay is neglecting their duties or non-existant you are probably not gonna be revived in the first place and don't have to worry about random traumas either way

Taking medication **will** be out of my control if
There are no chemists 
medbay staff are adamant on keeping non-medical personnel out of chemistry, assisted by security
someone destroys all the chem dispensers and engineers can't be bothered to replace them
the fucking room/dispensers are out of power
medbay is bombed

at least add Brendol to sleeper tier 2 and nanomeds

Cranizine requires the cooperation and competence of Botany, Chemistry and Science, and should have no side-effect for the fucking cost of making it

**Comment by FriendlyContractor** [comment]
Closing due to overwhelming negative feedback

**Comment by FriendlyContractor** [comment]
Closing due to overwhelming negative feedback
